blog/content/post/2022-01-03-pcie_screamer.md

```diff
@@ -111,108 +108,194 @@ To prepare `enjoy-digital/pcie_screamer`, the following steps were tried:
-     File "./pcie_screamer.py", line 160, in <module>
-       main()
-     File "./pcie_screamer.py", line 150, in main
-       prog.load_bitstream("build/gateware/top.bit")
-     File "/home/ibagnucki/git/pcie_screamer/litex/litex/litex/build/openocd.py", line 27, in load_bitstream
-       self.call(["openocd", "-f", config, "-c", script])
-     File "/home/ibagnucki/git/pcie_screamer/litex/litex/litex/build/generic_programmer.py", line 100, in call
-       raise OSError(msg)
-   OSError: Error occured during OpenOCD's call, please check:
-   - OpenOCD installation.
-   - access permissions.
-   - hardware and cable.
-   ```
-
-## Power problem
-
-Programming of the PCIeScreamer in the previous step failed. There could be
-several reasons why this could happen. It could be a broken programmer,
-misconfigured environment, or nonfunctional FPGA. The third option the problem
-with the FPGA seem to be most probable as
-[LambdaConcept's instructions](https://docs.lambdaconcept.com/screamer/getting_started.html#boot-the-target-system)
-warn about correct power delivery to the board and explain that the green diode
-LD3 should be on.
-
-The diode is indeed on, but only if JTAGSerial Programmer is connected to the
-power too. When the Programmer is disconnected, the led goes dark too. This LED
-behavior may (but not necessarily have to) indicate invalid power of the board.
-Another indication that something may be wrong with the PCIe connection is that
-no new PCIe device was shown on the tested machine using the `lspci` command.
-
-![](/img/PCIeScreamer_LED.jpg)
+This command will recursively build the entire project, starting with the 
+reconstruction of used IP Cores through the synthesis and implementation phase. 
+As a result, an FPGA configuration file will be created. Attention! this command
+may take up to an hour to execute (depending on the speed of the computer used).
+
+7. The last command is:
+   
+  ```tcl
+   vivado_flash_hs2.tcl 
+  ```
+  which is writing the configuration file (bitstream) to the FPGA board (using 
+  JTAG programmer/debugger). Attention! Before issuing the last command JTAG 
+  programmer (for example Xilinx Cable) must be properly connected to JTAG header 
+  on PCIScreamer board and it must be properly powered - see screen-shot:
+
+  ![Vivado TCL console](PCIScreamerBoard.jpg)
+In our case after writing bitstream file to Flash memory of FPGA board there were
+information in Vivado `TCL console` window that this operation was successful.
+
+## Testing PCIScreamer board in desktop computer
+After the correct programming of the PCIScreamer FPGA board, we proceeded to 
+testing its operation in a real computer. After embedding the PCIScreamer board
+in the PCI express slot and switching the power to PCI express, we started testing
+it in the Linux operating system (Ubuntu). The green LED was on, which indicated 
+that the power supply was correct and the board is working properly.Next step was
```

Comment:
```suggestion
that the power supply was correct and the board is working properly.The device is visible on PCI bus:
```

Please remove these lines, `lspci` is a commonly used command and there is no need to explain it
```
```bash
  $ lspci
``` 
This command list all devices seen on PCI bus in computer. Its result was:
```